packages/core/src/hooks/useSendTransaction.ts

```diff
@@ -0,0 +1,65 @@
+import { TransactionResponse, TransactionRequest } from '@ethersproject/abstract-provider'
+import { Signer } from 'ethers'
+import React, { useState } from 'react'
+import { StoredTransaction, useTransactionsContext } from '../providers'
+import { TransactionStatus } from '../../src'
+import { useEthers } from './useEthers'
+
+async function handleTransactionState(
+  transactionPromise: Promise<TransactionResponse>,
+  setState: React.Dispatch<React.SetStateAction<TransactionStatus>>,
+  addTransaction: (payload: StoredTransaction) => void,
+  chainId: number,
+  transactionName?: string
+) {
+  let transaction: TransactionResponse | undefined = undefined
+  try {
+    transaction = await transactionPromise
+
+    setState({ transaction, status: 'Mining', chainId })
+    addTransaction({
+      transaction,
+      submittedAt: Date.now(),
+      transactionName,
+    })
+
+    const receipt = await transaction.wait()
+
+    setState({ receipt, transaction, status: 'Success', chainId })
+  } catch (e) {
+    const errorMessage = e.reason ?? e.message
+    if (transaction) {
+      setState({ status: 'Fail', transaction, receipt: e.receipt, errorMessage, chainId })
+    } else {
+      setState({ status: 'Exception', errorMessage, chainId })
+    }
+  }
+}
+
+interface Options {
+  signer?: Signer
+  transactionName?: string
+}
```

Comment:
Let's extract common TransactionHookOptions and use the same for type for `useContractFunction`

Reply:
We can than reuse `connectContractToSigner` as well in both functions

Reply:
Idea: Can we extract on more generic function, i.e.:
```ts
useSendGenericTransaction(() => Promise<TransactionResponse>, options?: Options) {
  ...
}
```

So that useSendTransaction would be something like:
```ts
function useSendTransaction(...) {
  return useSendGenericTransaction(() => 
    signer.sendTransaction(transactionRequest)
  )
}
```